### ISSUE TITLE
Error in oil consumption effect on aircraft weight

### ISSUE THREAD

**Issue by dany93** [Juanvvc/c172p-detailed#847]
When I looked for giving a response to @wlbragg's question https://github.com/Juanvvc/c172p-detailed/issues/845 about the extra weight for 180 hp version, I was temporarily confused.
By checking the empty weight (no fuel, no pilot) in "Equipment > Fuel and Payload", I discovered that the values were different from the ones entered in the empty 160 hp FDM (1512 lbs instead of 1500 lbs in the FDM) or the FDM + extra weight due to 180 hp engine (1654 lbs instead of 1642 lbs in the FDM). The issue is that the engine oil content in FG has been added to the aircraft empty weight, in contradiction with the POHs where the "Basic Empty Weight" includes the full oil (section 6, Weight and Balance).
The issue is not really the absolute weight of the aircraft because these differences are weak and nobody will notice their effect. Moreover, the basic aircraft weight in the POHs are indicative, a pilot must know the true weight for his particular aircraft. **The issue is that, in FG, it is confusing when one searches to understand the files or to fix a bug**.
To be rigorous, the oil which has been consumed should be subtracted from the aircraft weight (instead of adding the remaining oil content).
But the difference is so weak that, in my opinion, the effect of oil consumption on the aircraft weight might be neglected. It would not be a noticeable fault and it would avoid adding useless complexity in the files.

**Comment by gilbertohasnofb** [comment]
Hmm, so shouldn't we make the engines weight 12 pounds less so that everything is fine? That would be the easier fix. For instance, you write that our 160 hp 1512 lbs instead of the 1500 lbs according to the POH. So if we make the engine weight 1488 lbs, that would be the weight without oil, and once in the sim it would weight between 1488 and 1500 (the latter if the oil is full). If so this looks like an easy fix, or am I missing something, Dany?

**Comment by gilbertohasnofb** [comment]
@dany93 nice catch by the way! Sharp eyes :smile:

**Comment by dany93** [comment]
That's an easy fix, but what I don't like is for the ones who will try to understand where the "Empty weight" values come from. Their first search will be in the POH's.
That a bit tweaky. Some comments can help, but doing this for such a particular point as the oil is weird.

How much would the error be if the consumed oil was not taken into account?

**Comment by dany93** [comment]
However, if you prefer, as the oil is the only such point in the POH, why not ???
With comments in the FDM in front of the empty weight.

**Comment by gilbertohasnofb** [comment]
@dany93 I would say we do the easy fix as well as adding comments to explain that the "empty" weight given by the POH does consider full oil level.

Also, we need to do a bit of maths here: the engines actually do have different oil capacities. So if a 160hp engine weights 1500 lbs WITH full oil (7 quarts), how much exactly should it weight without oil? Same for the 180hp, engine weights 1642 lbs WITH full oil (8 quarts), how much exactly should it weight without oil? 

The difference in both cases you gave above was 12 lbs, but that can't be correct then.

**Comment by dany93** [comment]
Yes for the +12 lbs and +14 lbs for 160 hp and 180 hp, I had forgotten this (re-eddited my previous post when you wrote).

**Comment by gilbertohasnofb** [comment]
Thanks, Dany! So in that case I believe this is what we need to use:

160hp with oil: 1500 lbs
160hp without oil: 1488 lbs
180hp with oil: 1642 lbs
180hp without oil: 1628 lbs
difference without oil: 140 lbs

So I can take a look at the xml files and fix this today.

**Comment by gilbertohasnofb** [comment]
@dany93 is the weight of the 180 hp engine defined only via this single line below from `bushkit.xml`?:

```
        <switch name="extra-weight-180hp">
            <default value="0"/>
            <test logic="AND" value="142">
```

Or is there some other file I should change? I changed the default engine value in the `c172p.xml` file and the 180hp in the `bushkit.xml` file.

**Comment by gilbertohasnofb** [comment]
If that's the single file, then I am done and ready to push and create a PR.

**Comment by gilbertohasnofb** [comment]
By the way, lines 17-23 in `c172p.xml` are:

```
        <!-- 
        For 180 hp, engine[1]:
        **Manually** set Internal Properties /controls/engines/engine[1]/magnetos = 3
        better to set original magnetos to 0 (key)
        do not use "s" (would start engine[0], 160 hp)
        start with controls/engines/engine[1]/starter = 1 then back to 0, or Ctrl-click on "starter = ", twice to stop starter
         -->
```

Is this true? I am quite sure we can use "s" to start the 180hp.

**Comment by dany93** [comment]
`Is this true? I am quite sure we can use "s" to start the 180hp.`
I totally ignore what it means, why this is here.

**Comment by dany93** [comment]
`Or is there some other file I should change? I changed the default engine value in the c172p.xml file and the 180hp in the bushkit.xml file.`
It's a bit old in my memory, but I think that it is enough.
You can check the result in "Fuel and payload", with no fuel, no pilot, with the 160 hp and 180 hp engines. You should find 1500 lbs and 1642 lbs (with full oil).

**Comment by gilbertohasnofb** [comment]
@dany93 thanks for all the replies.

>> Is this true? I am quite sure we can use "s" to start the 180hp.
> I totally ignore what it means, why this is here.

It's strange, because I just tested it again and that statement really doesn't make any sense. I will remove that comment from that file as I think it refers to some prior implementation

> You can check the result in "Fuel and payload"

Yes, of course! I tested it and it works like a charm, I tested for both engines both with full and empty oil levels, it's all good. I am pushing the PR in 5 min.
